### PR TITLE
Feat/po types

### PR DESCRIPTION
**Pull request by HazemKhaled**
## :memo: Description

1. Add more types like Bill, PurchaseOrderComment, PurchaseReceive, PurchaseTax
2. Add template to RoleUser
3. Make sure translation pipes accept only i18n languages

### :dart: Relevant issues
https://github.com/Knawat/sp-purchase-orders/pull/3

### :gem: Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### :scroll: Example code
```js

``` 

## :vertical_traffic_light: How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] run npm run build

## :checkered_flag: Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] **I have added tests that prove my fix is effective or that my feature works**
- [x] **New and existing unit tests pass locally with my changes**
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/purchase-order/order.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { PurchaseOrderItem } from "./item";
+import { PurchaseReceive } from "./receive";
+import { PurchaseTax } from "./tax";
+import { SalesOrder } from "../sales-order";
+
+export interface PurchaseOrder {
+  adjustment: number;
+  adjustmentDescription: string;
+  attention: string;
+  billedStatus: string;
```
[the remark this answers is not in the document]
Comment (cactusaurelius) [reply src/purchase-order/order.ts:13]: allowed invoice status [https://www.zoho.com/inventory/api/v1/#Invoices](zoho docs invoices)
sent, draft, overdue, paid, void, unpaid, partially_paid and viewed
I couldn't find a specific range of allowed values in Zoho docs. The available values are for listing POs

**Review by cactusaurelius — APPROVED** [review]

File: src/bill.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export interface Bill {
+  balance: number;
+  billId: string;
+  billNumber: string;
+  date: string;
```
[the remark this answers is not in the document]
Comment (HazemKhaled) [reply src/bill.ts:5]: done

File: src/purchase-order/order.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (HazemKhaled) [reply src/purchase-order/order.ts:13]: Will do it later, we need to fix from OMS first

File: src/purchase-order/order.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  bills: Bill[];
+  comments: PurchaseOrderComment[];
+  createdTime: string;
+  currencyCode: string;
+  currencySymbol: string;
+  date: string;
```
[the remark this answers is not in the document]
Comment (HazemKhaled) [reply src/purchase-order/order.ts:19]: done

File: src/purchase-order/order.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  deliveryAddress: Address;
+  deliveryCustomerId: string;
+  deliveryCustomerName: string;
+  deliveryDate: string;
```
[the remark this answers is not in the document]
Comment (HazemKhaled) [reply src/purchase-order/order.ts:23]: done

File: src/purchase-order/order.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  deliveryOrgAddressId: string;
+  exchangeRate: number;
+  expectedDeliveryDate: string;
```
[the remark this answers is not in the document]
Comment (HazemKhaled) [reply src/purchase-order/order.ts:26]: done

File: src/purchase-order/order.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  isDropShipment: boolean;
+  isInclusiveTax: boolean;
+  items: PurchaseOrderItem[];
+  lastModifiedTime: string;
```
[the remark this answers is not in the document]
Comment (HazemKhaled) [reply src/purchase-order/order.ts:30]: done

File: src/purchase-order/order.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  notes: string;
+  orderStatus: string;
+  purchaseorderId: string;
+  purchaseorderNumber: string;
+  purchasereceives: PurchaseReceive[];
+  receivedStatus: string;
```
[the remark this answers is not in the document]
Comment (HazemKhaled) [reply src/purchase-order/order.ts:36]: already string into OpenAPI

File: src/purchase-order/order.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  referenceNumber: string;
+  salesorderId: string;
+  salesorders: SalesOrder[];
+  shipVia: string;
+  shipmentTrackingNumber?: string;
+  status: string;
```
[the remark this answers is not in the document]
Comment (HazemKhaled) [reply src/purchase-order/order.ts:42]: can't get it from OpenAPI

File: src/purchase-order/receive.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import { Bill } from "bill";
+
+export interface PurchaseReceive {
+  bills: Bill[];
+  date: string;
```
[the remark this answers is not in the document]
Comment (HazemKhaled) [reply src/purchase-order/receive.ts:5]: done

**Pull request merged**